secrets/external.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func CheckPWCMD(pwCmd string) (string, error) {
+	if pwCmd == "" && os.Getenv("OS_PASSWORD") != "" {
+		return "", fmt.Errorf("no command specified")
```

Comment:
This shouldn't be an hard error. If no OS_PW_CMD env is defined then this function should act as a noop.